apps/client/src/app/auth/_state/server/hooks/usePostSignUp.ts

```diff
@@ -0,0 +1,33 @@
+import { useModal } from "@shophub/ui";
+import { useMutation } from "@tanstack/react-query";
+import type { SignUpFormValues } from "~/app/auth/_types";
+import { authApi } from "../api";
+
+export const usePostUserSignUp = () => {
+  const modal = useModal();
+
+  const { isLoading, mutateAsync } = useMutation(
+    async (userInfo: SignUpFormValues) => {
+      const res = await authApi.signUp(userInfo);
+
+      return res.data;
```

Comment:
#51 PR이 머지가 되면 이 부분도 수정이 되어야 하는데요, 51 PR의 res?.data.result 이 부분을 삭제하고

```suggestion
      const { data } = await authApi.signUp(userInfo);

      return data.result;
```

이렇게 하면 됩니다.